root/edit/details/EditArtist.js

```diff
@@ -0,0 +1,183 @@
+  +display_data: {
+    +area?: CompT<AreaT | null>,
+    +artist: ArtistT,
+    +begin_area?: CompT<AreaT | null>,
+    +begin_date?: CompT<PartialDateT | null>,
+    +comment?: CompT<string>,
+    +end_area?: CompT<AreaT | null>,
+    +end_date?: CompT<PartialDateT | null>,
+    +ended?: CompT<boolean>,
+    +gender?: CompT<GenderT | null>,
+    +ipi_codes?: CompT<$ReadOnlyArray<string> | null>,
+    +isni_codes?: CompT<$ReadOnlyArray<string> | null>,
+    +name: CompT<string>,
+    +sort_name?: CompT<string>,
+    +type?: CompT<ArtistTypeT | null>,
+  },
+}
+
+const EditArtist = ({edit}: {edit: EditArtistProps}) => {
+  const display = edit.display_data;
+  const area = display.area;
+  const beginDate = display.begin_date;
+  const beginArea = display.begin_area;
+  const endDate = display.end_date;
+  const endArea = display.end_area;
+  const ended = display.ended;
+  const gender = display.gender;
+  const ipiCodes = display.ipi_codes;
+  const isniCodes = display.isni_codes;
+  const type = display.type
+  let displayTypeId;
+  if (type && (type.new !== type.old)) {
+    displayTypeId = 0;
+  } else {
+    displayTypeId = display.artist.typeID;
+  }
+  return (
+    <>
+      <table className="details">
+        <tbody>
+          <tr>
+            <th>{addColonText(l('Artist'))}</th>
+            <td><EntityLink entity={display.artist} /></td>
+          </tr>
+        </tbody>
+      </table>
+      <table className="details edit-artist">
+        <tbody>
+          {display.name ? (
+            <Diff
+              label={addColonText(l('Name'))}
+              newText={display.name.new}
+              oldText={display.name.old}
+              split="\s+"
+            />
+          ) : null}
+          {display.sort_name ? (
+            <Diff
+              label={addColonText(l('Sort name'))}
+              newText={display.sort_name?.new ?? ""}
+              oldText={display.sort_name?.old ?? ""}
+              split="\s+"
+            />
+          ) : null}
+          {display.comment ? (
+            <Diff
+              label={addColonText(l('Disambiguation'))}
+              newText={display.comment?.new ?? ""}
+              oldText={display.comment?.old ?? ""}
+              split="\s+"
+            />
+          ) : null}
+          {type ? (
+            <FullChangeDiff
+              label={addColonText(l('Type'))}
+              newContent={type.new ? lp_attributes(type.new.name, 'artist_type') : ""}
+              oldContent={type.old ? lp_attributes(type.old.name, 'artist_type') : ""}
+            />
+          ) : null}
+          {gender ? (
+            <FullChangeDiff
+              label={addColonText(l('Gender'))}
+              newContent={gender.new?.name ? lp_attributes(gender.new.name, 'gender') : ""}
+              oldContent={gender.old?.name ? lp_attributes(gender.old.name, 'gender') : ""}
+            />
+          ) : null}
+          {area ? (
+            area.new?.gid === area.old?.gid ? null : (
```

Comment:
You don't have to check if the areas are different: the edits only store changed data, so they are *always* different if the field exists.